### DISCUSSION TITLE
Workaround for 3 second interaction timeout

### DISCUSSION THREAD

**Discussion by Dmitriy-ksm** [discord-net/Discord.Net#2732] [Ideas]
My bot works with an external api, and moreover, it processes images, so far I have not cached all the images I need and therefore sometimes slash commands fall due to the 3 second response time limit. In this regard, I propose the following feature.
Add the ability to provide some stub (text, or file) to the bot to respond after calling the slash command, which it will send to the chat while it processes the request. And after processing is completed, this message will be edited for the real result.
If you can't edit the submitted file due to discord restrictions, then for more flexibility, send the stub as an ephemeral message, and then send the real answer with another message.

**Comment by CottageDwellingCat** [comment]
You could call `DeferAsync()` do give your application up to 15 minutes to respond (using `FollowupAsync()`), or respond to the interaction then call `ModifyOrignalResponseAsync()` within 15 minutes, or respond to the interaction ephemerally, then followup. In ye olden days the library had a setting that defaulted to on called 'AlwaysAcknowledgeInteractions' which kind of did what you where talking about, but iirc that got removed before labs was merged into Discord.Net.

**Reply by Dmitriy-ksm** [reply]
Oh thanks, when I get some free time - I'll try your advice!

**Reply by Dmitriy-ksm** [reply]
Several additions. Didn't find ModifyOriginalResponseAsync. In general, all subsequent Followups change the response to the interaction. There is one peculiarity - the properties passed to DeferAsync are preserved for subsequent followups, so you cannot dynamically change ephemeral to true . I settled on the following solution. I call DeferAsync(ephemeral: true) followed by FollowupAsync("Your request in progress") and keep a link to this message. Later in the processing, I call Channel.SendFileAsync(file, fileName) and delete the reaction with DeleteAsync(). The disadvantage of this approach is that information about who called the command is lost, because the interaction itself is deleted.
